Not compiling with esp32

The examples of the library are not compiling with esp32.

It gives the following error:

```
In file included from C:\Users\Bernhard\Documents\Arduino\libraries\107-Arduino-UAVCAN\src/ArduinoUAVCAN.h:28:0,

                 from C:\Users\Bernhard\AppData\Local\Temp\arduino_modified_sketch_951541\UAVCAN-Heartbeat-Publish.ino:15:

C:\Users\Bernhard\Documents\Arduino\libraries\107-Arduino-UAVCAN\src/utility/convert.hpp:21:18: error: expected '{' before '::' token

 namespace arduino::_107_::uavcan

                  ^

C:\Users\Bernhard\Documents\Arduino\libraries\107-Arduino-UAVCAN\src/utility/convert.hpp:21:20: error: '_107_' in namespace '::' does not name a type

 namespace arduino::_107_::uavcan

                    ^

UAVCAN-Heartbeat-Publish:127:1: error: expected '}' at end of input

 }

 ^

exit status 1

expected '}' at end of input
```

To solve this error is somehow beyond my possibilities...

I did further testing. The above error message was using the latest release installed via Arduino Library Manager.

Now I updated to the master branch of github. This gives a slightly different error message but the program still does not compile:

```
In file included from C:\Users\Bernhard\Documents\Arduino\libraries\107-Arduino-UAVCAN\src/types/uavcan/node/ExecuteCommand.1.0.hpp:19:0,

                 from C:\Users\Bernhard\Documents\Arduino\libraries\107-Arduino-UAVCAN\src/ArduinoUAVCANTypes.h:15,

                 from C:\Users\Bernhard\Documents\Arduino\libraries\107-Arduino-UAVCAN\src/ArduinoUAVCAN.h:24,

                 from C:\Users\Bernhard\AppData\Local\Temp\arduino_modified_sketch_855220\UAVCAN-Heartbeat-Publish.ino:15:

c:\users\bernhard\documents\arduino\libraries\107-arduino-uavcan\src\utility\convert.hpp:21:18: error: expected '{' before '::' token

 namespace arduino::_107_::uavcan

                  ^

c:\users\bernhard\documents\arduino\libraries\107-arduino-uavcan\src\utility\convert.hpp:21:20: error: '_107_' in namespace '::' does not name a type

 namespace arduino::_107_::uavcan

                    ^

In file included from C:\Users\Bernhard\Documents\Arduino\libraries\107-Arduino-UAVCAN\src/ArduinoUAVCANTypes.h:15:0,

                 from C:\Users\Bernhard\Documents\Arduino\libraries\107-Arduino-UAVCAN\src/ArduinoUAVCAN.h:24,

                 from C:\Users\Bernhard\AppData\Local\Temp\arduino_modified_sketch_855220\UAVCAN-Heartbeat-Publish.ino:15:

C:\Users\Bernhard\Documents\Arduino\libraries\107-Arduino-UAVCAN\src/types/uavcan/node/ExecuteCommand.1.0.hpp: In member function 'void arduino::ExecuteCommand_1_0::Response::operator=(arduino::ExecuteCommand_1_0::Response::Status)':

C:\Users\Bernhard\Documents\Arduino\libraries\107-Arduino-UAVCAN\src/types/uavcan/node/ExecuteCommand.1.0.hpp:116:28: error: 'arduino::_107_' has not been declared

     data.status = arduino::_107_::uavcan::to_integer(status);

                            ^

In file included from C:\Users\Bernhard\Documents\Arduino\libraries\107-Arduino-UAVCAN\src/ArduinoUAVCANTypes.h:16:0,

                 from C:\Users\Bernhard\Documents\Arduino\libraries\107-Arduino-UAVCAN\src/ArduinoUAVCAN.h:24,

                 from C:\Users\Bernhard\AppData\Local\Temp\arduino_modified_sketch_855220\UAVCAN-Heartbeat-Publish.ino:15:

C:\Users\Bernhard\Documents\Arduino\libraries\107-Arduino-UAVCAN\src/types/uavcan/node/Heartbeat.1.0.hpp: In member function 'void arduino::Heartbeat_1_0::operator=(arduino::Heartbeat_1_0::Health)':

C:\Users\Bernhard\Documents\Arduino\libraries\107-Arduino-UAVCAN\src/types/uavcan/node/Heartbeat.1.0.hpp:78:34: error: 'arduino::_107_' has not been declared

     data.health.value = arduino::_107_::uavcan::to_integer(health);

                                  ^

C:\Users\Bernhard\Documents\Arduino\libraries\107-Arduino-UAVCAN\src/types/uavcan/node/Heartbeat.1.0.hpp: In member function 'void arduino::Heartbeat_1_0::operator=(arduino::Heartbeat_1_0::Mode)':

C:\Users\Bernhard\Documents\Arduino\libraries\107-Arduino-UAVCAN\src/types/uavcan/node/Heartbeat.1.0.hpp:83:32: error: 'arduino::_107_' has not been declared

     data.mode.value = arduino::_107_::uavcan::to_integer(mode);

                                ^

C:\Users\Bernhard\AppData\Local\Temp\arduino_modified_sketch_855220\UAVCAN-Heartbeat-Publish.ino: At global scope:

UAVCAN-Heartbeat-Publish:127:1: error: expected '}' at end of input

 }

 ^

Bibliothek SPI in Version 1.0 im Ordner: C:\Users\Bernhard\AppData\Local\Arduino15\packages\esp32\hardware\esp32\1.0.4\libraries\SPI  wird verwendet

Bibliothek 107-Arduino-UAVCAN in Version 1.1.0 im Ordner: C:\Users\Bernhard\Documents\Arduino\libraries\107-Arduino-UAVCAN  wird verwendet

Bibliothek 107-Arduino-MCP2515 in Version 1.2.2 im Ordner: C:\Users\Bernhard\Documents\Arduino\libraries\107-Arduino-MCP2515  wird verwendet

exit status 1

expected '}' at end of input

```

--- a/examples/UAVCAN-Blink/UAVCAN-Blink.ino
+++ b/examples/UAVCAN-Blink/UAVCAN-Blink.ino
@@ -30,6 +30,10 @@ static int const MKRCAN_MCP2515_CS_PIN  = 3;
 static int const MKRCAN_MCP2515_INT_PIN = 7;
 static CanardPortID const BIT_PORT_ID   = 1620U;
 
+#if defined(ESP32)
+static int const LED_BUILTIN = 2;
+#endif
+
 /**************************************************************************************
  * FUNCTION DECLARATION
  **************************************************************************************/

--- a/examples/UAVCAN-Heartbeat-Subscribe/UAVCAN-Heartbeat-Subscribe.ino
+++ b/examples/UAVCAN-Heartbeat-Subscribe/UAVCAN-Heartbeat-Subscribe.ino
@@ -114,7 +114,7 @@ void onHeartbeat_1_0_Received(CanardTransfer const & transfer, ArduinoUAVCAN & /
   char msg[64];
   snprintf(msg, 64,
            "ID %02X, Uptime = %d, Health = %d, Mode = %d, VSSC = %d",
-           transfer.remote_node_id, hb.data.uptime, hb.data.health, hb.data.mode, hb.data.vendor_specific_status_code);
+           transfer.remote_node_id, hb.data.uptime, hb.data.health.value, hb.data.mode.value, hb.data.vendor_specific_status_code);
 
   Serial.println(msg);
 }

--- a/src/utility/CritSec-esp32.c
+++ b/src/utility/CritSec-esp32.c
@@ -1,0 +1,38 @@
+/**
+ * This software is distributed under the terms of the MIT License.
+ * Copyright (c) 2020 LXRobotics.
+ * Author: Alexander Entinger <alexander.entinger@lxrobotics.com>
+ * Contributors: https://github.com/107-systems/107-Arduino-UAVCAN/graphs/contributors.
+ */
+
+/**************************************************************************************
+ * INCLUDE
+ **************************************************************************************/
+
+#include "CritSec.h"
+
+#ifdef ARDUINO_ARCH_ESP32
+
+#include <Arduino.h>
+
+/**************************************************************************************
+ * GLOBAL VARIABLES
+ **************************************************************************************/
+
+static portMUX_TYPE mtx = portMUX_INITIALIZER_UNLOCKED;
+
+/**************************************************************************************
+ * FUNCTION DEFINITION
+ **************************************************************************************/
+
+void crit_sec_enter()
+{
+  portENTER_CRITICAL(&mtx);
+}
+
+void crit_sec_leave()
+{
+  portEXIT_CRITICAL(&mtx);
+}
+
+#endif /* ARDUINO_ARCH_ESP32 */

--- a/src/utility/convert.hpp
+++ b/src/utility/convert.hpp
@@ -18,7 +18,11 @@
  * NAMESPACE
  **************************************************************************************/
 
-namespace arduino::_107_::uavcan
+namespace arduino
+{
+namespace _107_
+{
+namespace uavcan
 {
 
 /**************************************************************************************
@@ -35,6 +39,8 @@ constexpr auto to_integer(Enumeration const value) -> typename std::underlying_t
  * NAMESPACE
  **************************************************************************************/
 
-} /* arduino::_107_::uavcan */
+} /* uavcan */
+} /* _107_ */
+} /* arduino */
 
 #endif /* ARDUINO_UAVCAN_UTILITY_COVERT_HPP_ */

